src/provider/provider_tracking.c

```diff
@@ -233,6 +219,10 @@ umf_memory_provider_ops_t UMF_TRACKING_MEMORY_PROVIDER_OPS = {
 umf_result_t umfTrackingMemoryProviderCreate(
     umf_memory_provider_handle_t hUpstream, umf_memory_pool_handle_t hPool,
     umf_memory_provider_handle_t *hTrackingProvider) {
+#if defined(_WIN32) && !defined(UMF_SHARED_LIBRARY)
```

Comment:
nit: I think it might be better to always implement `umfTrackingMemoryProviderInit` and just do nothing there if we're on linux or we build a static lib. This way we don't have to duplicate this condition averywhere.